### PR TITLE
Generate ENI using Python script

### PR DESCRIPTION
**Pull request by wupeka**
# Description of change

/e/n/i is now generated using a basic template and a python
script that fills the template based on information obtained
from ip command. This new approach removes dependency from
interface names that are, in newer ubuntu releases, decided on
boot time.
The networking is disabled in cloud-init using network: config: "disabled"
Notes:
- network:config:disabled method doesn't work on cloud-init used in Trusty, the only downside is that the network will be started using defaults (dhcp) and then shut down when we rewrite the e/n/i file. 
- there's a bug in cloud-init on Xenial that causes an error in log message: https://bugs.launchpad.net/cloud-init/+bug/1679602 , it doesn't cause any functional problems though. 

# QA Steps

Bootstrap juju and add-machine for a kvm of the latest ubuntu version, there should be a /e/n/i file comprising all the interfaces with their correct names, interfaces.bak with old file and interfaces.templ with the template used to generate ENI.

# Documentation changes

None

# Bug reference

https://bugs.launchpad.net/juju/+bug/1666198

### CODE REVIEW

**Comment by wupeka** [comment]
!!test!!

**Comment by wupeka** [comment]
!!retest!!

**Comment by wupeka** [comment]
!!testme!!

**Comment by wupeka** [comment]
!!runthetests!!

**Comment by wupeka** [comment]
!!test!!

**Comment by wupeka** [comment]
!!testplease!!

**Comment by wupeka** [comment]
!!pleasetest!!

**Comment by wupeka** [comment]
!!testme!!

**Comment by wupeka** [comment]
!!test!!

**Comment by wupeka** [comment]
!!retest!!

File: container/lxd/lxd.go [hunk 1]
```diff
@@ -165,29 +166,18 @@ func (manager *containerManager) CreateContainer(
...
-		return
-	}
-
 	spec := lxdclient.InstanceSpec{
 		Name:     name,
 		Image:    imageName,
 		Metadata: metadata,
 		Devices:  nics,
 		Profiles: profiles,
+		// We overwrite the (default) eth0.cfg file so that the trusty container
```
Comment (jameinel) [inline container/lxd/lxd.go]: This didn't seem to do what you have commented here. I still see on LXD the container coming up with "listening on LPF/eth0/..." and and then ifdown stopping that and doing DHCPRELEASE. I have the feeling that cloud-init is creating the file always on Trusty, so it doesn't help us to overwrite it. Have you tried without this and seen if it acts differently?

Comment (wupeka) [inline container/lxd/lxd.go]: That was an artifact from trying to turn off cloud-init networking completely, removed (along with Files functionality for LXD as it was the only place it's used)

**Comment by wupeka** [comment]
!!pleasetestthischange!!

**Comment by wupeka** [comment]
$$merge$$

**Pull request merged**